GameEngine/DeterministicDeck.cs

```diff
@@ -0,0 +1,92 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Linq;
+
+namespace GameEngine
+{
+    public class DeterministicDeck : Deck
+    {
+        public List<CardType> cards { get; private set; }
+        public override List<CardType> Cards => cards;
+        public override int Count => Cards.Count;
+
+        private DeterministicDeck() { }
```

Comment:
💡 The existence of a public constructor that takes more than 1 argument means you do not need to explicitly define the private default constructor

Reply:
This constructor is used in the `Clone` method.